articles/node-url-shortener/index.md

```diff
@@ -0,0 +1,231 @@
+1. A basic understanding of the [JavaScript](https://www.w3schools.com/js/DEFAULT.asp) programming language.
+   
+2. A basic understanding of [Node.js](https://nodejs.org/en/) and [Express](https://expressjs.com/) framework.
+   
+3. Have [Postman](https://www.postman.com/downloads/) HTTP API client installed on your system.
+   
+4. A text editor, preferably [VS Code](https://code.visualstudio.com/download).
+
+5. [MongoDB](https://www.mongodb.com/try/download/community) database server installed on your system.
+
+### Setting Up The Project
+Create a folder named `URL-Shortener-Service` and open it on your favorite IDE. I will be using VS Code here. Go inside the folder and type `npm init` to generate an initial package for our project. This command will require some prompts, to skip this, you can use `npm init -y`. Our project will use various npm packages from the npm registry. The npm packages include:
+
+- `express`: Express is a backend web application framework for Node.js for building web applications and APIs.
+
+- `mongoose`: Mongoose is an asynchronous database driver or Object Data Mapper for MongoDB. It is responsible for connecting to the database and performing query operations.
+
+- `short-id`: The `short-id` module creates user-friendly and unique ids for our URLs.
+
+- `valid-url`: This is a module that verifies all the URLs sent to the API.
+
+- `nodemon`: nodemon package will be installed as a development dependency. It will constantly monitors our application by automatically restarting the server when any file changes.
+  
+Next, we now need to download the packages. Run the following command on your terminal to download them:
+
+```bash
+npm install express mongoose shortid valid-url
+```
+
+This will download the named packages inside `node_modules` folder and update the `package.json` file with the dependencies.
+
+>>> Note: `nodemon` is a development dependency. To install this, run the command `npm install --save-dev nodemon`.
+
+
+### Setting Up Express server
+Inside our `URL-Shortener-Service` folder, create a file named `server.js`. The following is the initial code to start the Express server:
+
+```js
+    const express = require('express')
+
+    const app = express()
+
+    const PORT = process.env.PORT || 5000
+
+    app.listen(PORT, ()=> console.log('DB connection established'))
+```
+
+This starter code imports the express package. The `app = express()` creates an instance of our application.
+In our app, we need to listen to the incoming request. `app.listen` takes a port number and a callback function that is invoked upon a successful connection.
+
+### Configuring the MongoDB connection
+We will use the `mongoose` package that we installed via npm as the database driver. To configure the database, create a `config` folder inside our `URL-Shortener-Service` folder. Inside the `config` folder, add a file named `db.config.js`. This is the file to add the following database connection code:
+
```

Comment:
Please add a heading that mentions that this code is to be included inside `db.config.js`